src/modules/user/user.usecase.ts

```diff
@@ -993,6 +993,102 @@ export class UserUseCases {
     await this.keyServerRepository.deleteByUserId(user.id);
   }
 
+  async recoverAccountLegacy(
+    userUuid: User['uuid'],
+    credentials: LegacyRecoverAccountDto,
+  ): Promise<void> {
+    const { mnemonic, password, salt, asymmetricEncryptedMnemonic } =
+      credentials;
+
+    const user = await this.userRepository.findByUuid(userUuid);
+
+    if (!user) {
+      throw new NotFoundException('User not found');
+    }
+
+    await this.userRepository.updateByUuid(userUuid, {
```

Comment:
I would suggest to do this as the last step. Why? Because if the steps after this one fail, and even if the API returns an error, the door for signing in successfully when not everything it's okay will be open for the user.

As you already updated the password, the user could access to the account but its keys would be wrong. As this is a recover account operation, the user already cannot access to it, so it would be better to do this as the last step, as the user could sign in again only after completing every step successfully.